sdonboarding.client/src/components/UITemplate/EditDataTemplate.jsx

```diff
@@ -0,0 +1,284 @@
+                                            onChange={(e) => {
+                                                const selectedOption = e.target.selectedOptions[0];   
+                                                handleListChange(selectedOption.dataset.id, selectedOption.value, selectedOption.dataset.type);
+                                            }}
+                                            className="w-full px-4 py-2 border rounded text-gray-500"
+                                        >
+                                            {localStores && localStores.map((store) => (
+                                                <option key={store.id}
+                                                    value={store.name}
+                                                    data-id={store.id}                  // Custom data attribute for store ID, these will get passed to handleListChange
+                                                    data-type='Store'
+                                                >
+                                                    {store.name}
+                                                </option>))}
+
+                                        </select>
+                                    ) : formElementType[index] === 'date' && !isNaN(Date.parse(localObject[key])) &&
+                                         (localObject[key].includes('-') || localObject[key].includes('/')) ? (
+                                         // Render date input and removing the T part
+                                          < input
+                                            type="date"
+                                            name={ key }
+                                            id={ key }
+                                            value={localObject[key].split('T')[0]}
+                                            onChange={ handleInputChange }
+                                            className="w-full px-4 py-2 border rounded text-gray-500"
+                                            />
+                                    ) : (
+                                          // Render text input for other fields (like 'name', 'price')
+                                        <input
+                                            type="text"
+                                            name={key}
+                                            id={key}
+                                            value={localObject[key]}
+                                            onChange={handleInputChange}
+                                            className="w-full px-4 py-2 border rounded text-gray-500"
+                                            placeholder={`Enter ${labels[key]}`}
+                                        />
+                                    )}
+                                </div>
+                            );
+                        })}
```

Comment:
Again, this is too long and hard to read. Please check my previous comment about this one as well